temoa/model_checking/unit_checking/relations_checker.py

```
@@ -0,0 +1,518 @@
+                'having commodity field--expecting tech field.  Check data.'
+            )
+        query = (
+            f'SELECT {ct.commodity_reference if ct.commodity_reference else "tech"}, '
+            f'units FROM {ct.table_name}'
+        )
+        try:
+            rows = conn.execute(query).fetchall()
+        except sqlite3.OperationalError:
+            logger.error(
+                'failed to process query: %s when processing table %s', query, ct.table_name
+            )
```

Ruff (0.14.7)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)